### DISCUSSION TITLE
无法创建共享计算组副本

### DISCUSSION THREAD

**Discussion by coulsonm** [ByConity/ByConity#579] [Q&A] [answered]
**创建计算组：**
CREATE WAREHOUSE
IF NOT EXISTS `vw_default`
SETTINGS num_workers = 1, type = 'Default',
         auto_suspend = 3600, auto_resume = 1,
         min_worker_groups = 0, max_worker_groups = 1, max_concurrent_queries=200;

**创建计算组副本：**
CREATE WORKER GROUP
IF NOT EXISTS `wg_default` in `vw_default`
SETTINGS type = 'Physical';

**向计算组中添加机器：**
ByConity 创建计算组和计算组副本的元信息后，则可以创建对应的 K8s Worker 资源，为 worker 配置对应的环境变量 VIRTUAL_WAREHOUSE_ID=$VW_NAME, WORKER_GROUP_ID=$WG_NAME。Worker 启动后即可自动注册到对应的 virtual warehouse 和 worker group 中。

这里就有一个问题，计算组副本是启动worker时添加的，只能指定一个计算组副本，那么共享计算组副本要怎么添加呢？

**Comment by zeromem** [comment, marked as answer]
Hi @coulsonm 共享计算组副本是为了实现跨VW资源共享，在VW内有空闲WG的情况下，可以把自己的资源共享给其他VW使用。
首先，每个表都需要指定自己的读写VW（默认为cnch_vw_default = 'vw_default', cnch_vw_write = 'vw_write'）。然后可以通过在VW内创建共享WG的方式来实现共享。

举例：
已有VW：vw-tnt1-default， 包含一个WG：wg-tnt1-default。另一个VW：vw-tnt2-default，没有物理WG。

表a参数为cnch_vw_default='vw-tnt1-default'，则查询时将使用wg-tnt1-default。
假如现在wg-tnt1-default资源足够空闲，且希望共享给其他业务，比如表b（表b已经设置了自己的cnch_vw_default为 vw-tnt2-default）。

则我们可以在vw-tnt2-default中创建一个共享计算组副本，让其指向wg-tnt1-default：
```
CREATE WORKER GROUP IF NOT EXISTS `wg-tnt2-shared-default` in `vw-tnt2-default` SETTINGS type = 'Shared', shared_worker_group = 'wg-tnt1-default' ...
```
这样，vw-tnt2-default中新建了一个wg  wg-tnt2-shared-default，并且可以供表b查询时使用。它没有对应的实际物理资源，实际使用的是wg-tnt1-default
